samples/net/lwm2m_client/README.rst

```diff
@@ -6,14 +6,144 @@ LWM2M client
 Overview
 ********
 
-TODO
+Lightweight Machine to Machine (LwM2M) is an application layer protocol
+based on CoAP/UDP. and is designed to expose various resources for reading,
+writing and executing via an LwM2M server in a very lightweight environment.
```

Comment:
Actually, "an" is correct here.  Even though the first letter of LwM2M is a consonant,  the rule is whether the word starts with a consonant *sound* or not.  For example, it's "an hour", and "an honorable man", but "a horrible movie".